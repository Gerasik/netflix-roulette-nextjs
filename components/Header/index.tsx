import React, { FunctionComponent } from 'react';

import Logo from '../../components/Logo';
import styles from './style.module.scss';
import headerIMG from '../../assets/header_background.jpg';

const Header: FunctionComponent = ({ children }) => {
  return (
    <div className={styles['header-background']}>
      <img src={headerIMG} alt="posters" className={styles['background-image']} />
      <Logo />
      <div className={styles.container}>{children}</div>
    </div>
  );
};

export default Header;
