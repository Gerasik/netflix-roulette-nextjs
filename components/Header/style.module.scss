@import '../../styles/colors.scss';

.header-background {
  background: rgba($color: #000000, $alpha: 0.7);
  max-height: 20em;
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: center;
  position: relative;
  overflow-y: hidden;
  .container {
    width: 80%;
    padding: 2em 0;
  }

  .background-image {
    height: 100%;
    position: absolute;
    width: 100%;
    z-index: -1;
    object-fit: cover;
    filter: blur(0.2em);
  }
}
