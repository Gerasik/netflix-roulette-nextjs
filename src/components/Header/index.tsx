import React from 'react';
import _ from 'lodash';

import { useRouter } from 'next/router';
import Logo from 'components/Logo';
import styles from './style.module.scss';
import headerIMG from 'assets/header_background.jpg';

const Header = ({ children }) => {
  const router = useRouter();
  const arr = router.asPath.split('/') || null;

  return (
    <div className={styles['header-background']}>
      <img src={headerIMG} alt="posters" className={styles['background-image']} />
      <nav className={styles['header-nav']}>
        <Logo />
        {!arr.some(item => item === 'search') ? (
          <button onClick={() => router.back()} className={styles['search-icon']} />
        ) : null}
      </nav>
      <div className={styles.container}>{children}</div>
    </div>
  );
};

Header.getInitialProps = props => {
  console.log(props);
};

export default Header;
