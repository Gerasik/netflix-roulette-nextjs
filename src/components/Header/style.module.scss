@import 'styles/colors.scss';

.header-background {
  background: rgba($color: #000000, $alpha: 0.7);
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: center;
  position: relative;
  overflow-y: hidden;
  .container {
    width: 100%;
    padding: 2em 4em;
    box-sizing: border-box;
  }

  .background-image {
    height: 100%;
    position: absolute;
    width: 100%;
    z-index: -1;
    object-fit: cover;
    filter: blur(0.2em);
  }

  .search-icon {
    margin: 0.7em 1.2em;
    width: 1em;
    height: 1em;
    border-radius: 50%;
    border: 0.1em solid $secondaryFontColor;
    position: relative;
    cursor: pointer;
    background: inherit;
    &:before {
      content: '';
      position: absolute;
      width: 0.1em;
      height: 0.6em;
      bottom: -0.3em;
      left: -0.3em;
      transform: rotate(52deg);
      background-color: $secondaryFontColor;
    }
  }
}
.header-nav {
  width: 100%;
  display: flex;
  justify-content: space-between;
  align-items: center;
}
