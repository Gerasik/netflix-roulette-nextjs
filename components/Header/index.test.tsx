import React from 'react';
import { shallow } from 'enzyme';

import Header from './index';

describe('<Header />', () => {
  it('renders an `img`', () => {
    const wrapper = shallow(<Header />);
    expect(wrapper.find('img').length).toBe(1);
  });

  it('renders children when passed in', () => {
    const wrapper = shallow(
      <Header>
        <div className="unique" />
      </Header>
    );
    expect(wrapper.contains(<div className="unique" />)).toBeTruthy();
  });

  it('renders children when passed in', () => {
    const wrapper = shallow(
      <Header>
        <div className="unique" />
      </Header>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
