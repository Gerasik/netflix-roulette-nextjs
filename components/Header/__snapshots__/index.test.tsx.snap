// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders children when passed in 1`] = `
<div
  className="header-background"
>
  <img
    alt="posters"
    className="background-image"
    src="header_background.jpg"
  />
  <Logo />
  <div
    className="container"
  >
    <div
      className="unique"
    />
  </div>
</div>
`;
